diff --git a/src/Plugin/Filter/CollapseText.php b/src/Plugin/Filter/CollapseText.php
index 32f0ae4..648ac3e 100644
--- a/src/Plugin/Filter/CollapseText.php
+++ b/src/Plugin/Filter/CollapseText.php
@@ -17,7 +17,7 @@ define('COLLAPSE_TEXT_DEFAULT_TITLE', t('Click here to expand or collapse this s
  *   id = "filter_collapse_text",
  *   title = @Translation("Collapsible text blocks"),
  *   description = @Translation("Allows the creation of collapsing blocks of text. This filter must be after the 'Limit allowed HTML tags' filter, and should be after the 'Convert line breaks into HTML' filter."),
- *   type = Drupal\filter\Plugin\FilterInterface::TYPE_MARKUP_LANGUAGE,
+ *   type = Drupal\filter\Plugin\FilterInterface::TYPE_TRANSFORM_IRREVERSIBLE,
  * )
  */
 class CollapseText extends FilterBase {
